### PR TITLE
KAZOO-5205 extend auth capabilities

### PR DESCRIPTION
**Pull request by lazedo**

### CODE REVIEW

File: core/kazoo_auth/src/kz_auth_profile.erl [hunk 1]
```diff
@@ -164,48 +175,103 @@ maybe_add_user_email(#{auth_provider := #{name := Prov}}=Token) ->
...
     lager:debug("identity not set, skip adding user"),
     Token.
 
+-spec ensure_profile_properties(ne_binary(), kz_proplist(), kz_proplist(), map()) -> map().
+ensure_profile_properties(DocId, Missing, Props, #{} = Token) ->
+    case kz_datamgr:open_cache_doc(?KZ_AUTH_DB, DocId) of
+        {'ok', Doc} ->
+            case Missing -- kz_json:get_keys(Doc) of
+                [] -> do_update_user(DocId, Props, Token);
+                StillMissing ->
```
Comment (fenollp) [inline core/kazoo_auth/src/kz_auth_profile.erl]: `_StillMissing`

File: applications/crossbar/doc/auth.md [hunk 2]
```diff
@@ -6,6 +6,36 @@
...
+#### Fetch
+
+> GET /v2/auth/links
+
+```shell
+curl -v -X GET \
+    -H "X-Auth-Token: {AUTH_TOKEN}" \
+    http://{SERVER}:8000/v2/auth/links
+```
+
```
Comment (fenollp) [inline applications/crossbar/doc/auth.md]: Someone stole the docs!

File: applications/crossbar/src/modules/cb_auth.erl [hunk 3]
```diff
@@ -115,12 +124,31 @@ validate_path(Context, ?CALLBACK_PATH, ?HTTP_PUT) ->
...
+validate_path(Context, ?LINKS_PATH, ?HTTP_POST) ->
+    cb_context:validate_request_data(<<"auth.callback">>, Context, fun maybe_link/1);
+
+validate_path(Context, ?LINKS_PATH, ?HTTP_DELETE) ->
+    cb_context:validate_request_data(<<"auth.callback">>, Context, fun maybe_unlink/1).
 
-validate_token(Context, Token) ->
-    case crossbar_auth:validate_auth_token(Token) of
-        {'error', Error} -> crossbar_util:response('error', Error, 401, Context);
+validate_token_info(Context, Token) ->
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_auth.erl]: spec

File: core/kazoo_auth/src/kz_auth_profile.erl [hunk 4]
```diff
@@ -164,48 +175,103 @@ maybe_add_user_email(#{auth_provider := #{name := Prov}}=Token) ->
...
+    Props = format_user_doc(Token),
+    case updates_needed(JObj, Props) of
+        [] -> maybe_required_properties_missing(Token, Props, JObj);
         Updates -> update_user(DocId, Updates, Token)
     end.
 
--spec updates_needed(kz_json:object(), map()) -> kz_proplist().
-updates_needed(JObj, Token) ->
-    Props = format_user_doc(Token),
+maybe_required_properties_missing(#{auth_provider := #{profile_required_props := RequiredProps}
```
Comment (jamesaimonetti) [inline core/kazoo_auth/src/kz_auth_profile.erl]: spec

**Pull request merged**